lib/std/jsxmlhttprequest.nim

```diff
@@ -0,0 +1,57 @@
+## `XMLHttpRequest` for the JavaScript target: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest
+when not defined(js):
+  {.fatal: "Module jsxmlhttprequest is designed to be used with the JavaScript backend.".}
+
+from std/dom import Node
+
+type XMLHttpRequest* = ref object of JsRoot  ## https://xhr.spec.whatwg.org
+  responseXML*: Node
+  withCredentials*: bool
+  status*, timeout*, readyState*: cint
+  responseText*, responseURL*, statusText*: cstring
+
+func newXMLHttpRequest*(): XMLHttpRequest {.importjs: "new XMLHttpRequest()".}
+  ## Constructor for `XMLHttpRequest`.
+
+func open*(this: XMLHttpRequest; metod, url: cstring; async = true; user = cstring.default; password = cstring.default) {.importjs: "#.$1(#, #, #, #, #)".}
```

Comment:
These are all **not**  funcs according to the "strict funcs" definition as the object the ``this`` pointer points to is mutated. Please change them all to ``proc``.

Reply:
Very true